pkg/api/types.go

```diff
@@ -1895,6 +1901,10 @@ type DeleteOptions struct {
 	// The value zero indicates delete immediately. If this value is nil, the default grace period for the
 	// specified type will be used.
 	GracePeriodSeconds *int64 `json:"gracePeriodSeconds"`
+
+	// Must be fulfilled before a deletion is carried out. If not possible, a 409 Conflict status will be
+	// returned.
+	Preconditions Preconditions `json:"preconditions"`
```

Comment:
I think it won't actually work right with omit empty unless it's a pointer. (Like, it won't actually omit empty structs.)